src/plugins.c

```diff
@@ -1186,16 +1495,89 @@ static void pm_plugin_toggled(GtkCellRendererToggle *cell, gchar *pth, gpointer
 			keybindings_load_keyfile();		/* load shortcuts */
 
 		/* update model */
-		gtk_list_store_set(pm_widgets.store, &store_iter,
+		gtk_tree_store_set(pm_widgets.store, &store_iter,
 			PLUGIN_COLUMN_CHECK, state,
 			PLUGIN_COLUMN_PLUGIN, p, -1);
 
 		/* set again the sensitiveness of the configure and help buttons */
 		pm_update_buttons(p);
+
+		/* Depending on the state disable the checkbox for the proxy of this plugin, and
+		 * only re-enable if the proxy is not used by any other plugin */
+		if (p->proxy != &builtin_so_proxy_plugin)
+		{
+			GtkTreeIter parent;
+			gboolean can_uncheck;
+			GtkTreePath *store_path = gtk_tree_model_filter_convert_path_to_child_path(
+			                                GTK_TREE_MODEL_FILTER(model), path);
```

Comment:
The rest of the function assumes the same. I think it's nearly impossible that it changed, since the tree was just created when opening the PM dialog (i.e. it's not cached from the startup).